libminifi/src/provenance/ProvenanceTaskReport.cpp

```diff
@@ -0,0 +1,221 @@
+	    logger_->log_error("Site2Site bootstrap failed yield period %d peer ",
+	                       processor->getYieldPeriodMsec());
+	    return;
+	}
+
+	int batch = 100;
+
+	if (context->getProperty(batchSize.getName(), value) && core::Property::StringToInt(value, lvalue))
+	{
+		batch = (int) lvalue;
+	}
+	
+	std::vector<std::shared_ptr<ProvenanceEventRecord>> records;
+	std::shared_ptr<ProvenanceRepository> repo = std::static_pointer_cast<ProvenanceRepository> (context->getProvenanceRepository());
+
+	repo->getProvenanceRecord(records, batch);
+
+	if (records.size() <= 0)
+	{
+		returnProtocol(std::move(protocol_));
+		return;
+	}
+
+	Json::Value array;
+	for (auto record : records)
+	{
+		Json::Value recordJson;
+		Json::Value updatedAttributesJson;
+		Json::Value parentUuidJson;
+		Json::Value childUuidJson;
+		recordJson["eventId"] = record->getEventId().c_str();
+		recordJson["eventType"] = ProvenanceEventRecord::ProvenanceEventTypeStr[record->getEventType()];
+		recordJson["timestampMillis"] = record->getEventTime();
+		recordJson["durationMillis"] = record->getEventDuration();
+		recordJson["lineageStart"] = record->getlineageStartDate();
+		recordJson["details"] = record->getDetails().c_str();
+		recordJson["componentId"] = record->getComponentId().c_str();
+		recordJson["componentType"] = record->getComponentType().c_str();
+		recordJson["entityId"] = record->getFlowFileUuid().c_str();
+		recordJson["entityType"] = "org.apache.nifi.flowfile.FlowFile";
+		recordJson["entitySize"] = record->getFileSize();
+		recordJson["entityOffset"] = record->getFileOffset();
+
+		for (auto attr : record->getAttributes())
+		{
+			updatedAttributesJson[attr.first] = attr.second;
+		}
+		recordJson["updatedAttributes"] = updatedAttributesJson;
+
+		for (auto parentUUID : record->getParentUuids())
+		{
+			parentUuidJson.append(parentUUID.c_str());
+		}
+		recordJson["parentIds"] = parentUuidJson;
+
+		for (auto childUUID : record->getChildrenUuids())
+		{
+			childUuidJson.append(childUUID.c_str());
+		}
+		recordJson["childIds"] = childUuidJson;
+		recordJson["transitUri"] = record->getTransitUri().c_str();
+		recordJson["remoteIdentifier"] = record->getSourceSystemFlowFileIdentifier().c_str();
+		recordJson["alternateIdentifier"] = record->getAlternateIdentifierUri().c_str();
+		recordJson["application"] = "MiNiFi Flow";
+		array.append(recordJson);
+	}
+
+	Json::StyledWriter writer;
+	std::string jsonStr = writer.write(array);
+	uint8_t *payload = (uint8_t *) jsonStr.c_str();
+	int length = jsonStr.length();
+
+	try
+	{
+		std::map<std::string, std::string> attributes;
+		protocol_->transferBytes(context, session, payload, length, attributes);
+	}
+	catch (...)
+	{
+		// if transfer bytes failed, return instead of purge the provenance records
+		returnProtocol(std::move(protocol_));
+		return;
+	}
+
+	// we transfer the record, purge the record from DB
+	repo->purgeProvenanceRecord(records);
+
+	returnProtocol(std::move(protocol_));
+
+	return;
```

Comment:
I'm sorry I don't follow. an empty return on a void isn't necessary. It's unnecessary code.